### PR TITLE
FB Pixel: Custom Content Types and Automatic Configuration

### PR DESCRIPTION
**Pull request by gpsamson**
**What does this PR do?**
- Adds a setting to disable [Automatic Configuration](https://developers.facebook.com/docs/facebook-pixel/api-reference#automatic-configuration).
- Adds a mapping for Content Types ([FB Pixel Settings](https://app.segment.com/partner-portal/integration/facebook-pixel/settings)). For backwards compatibility we are falling back to `product` and `product_group`.

**Are there breaking changes in this PR?**
Negative.

**Has this been tested end-to-end? Please provide screenshots on how the fix now populates in the end tool. If not, what was done to test?**
TODO in staging.

Local `automaticConfiguration` testing was verified in dev tools.

Local testing where a custom content type is passed to FB:
screen shot 2018-09-12 at 7 10 37 pm
The category "Games" was mapped to "vehicles" in this example.

**Does this require a metadata change? If so, please link the PR from https://github.com/segmentio/destination-scripts.**
Two new [settings](https://app.segment.com/partner-portal/integration/facebook-pixel/settings) were added, but migrations weren't needed.

**What are the relevant tickets?**
[PLATFORM-2538](https://segment.atlassian.net/browse/PLATFORM-2538)
[PLATFORM-2310](https://segment.atlassian.net/browse/PLATFORM-2310)

**Link to CC ticket**
https://segment.atlassian.net/browse/CC-1573

**List all the tests accounts you have used to make sure this change works**
Personal FB Pixel account

**Helpful Docs**
https://paper.dropbox.com/folder/show/Facebook-Pixel-e.1gg8YzoPEhbTkrhvQwJ2zz409feT37UwVJoNokfnCOndQwCZywN3

### CODE REVIEW

**Comment by jlee9595** [comment]
having a setting to explicitly 'disable autoconfig' sounds unnecessarily confusing to me. would it be possible to just have the setting be 'autoconfig', which can be disabled or enabled? i feel that would eliminate some redundancy there and make things more straightforward.

**Comment by gpsamson** [comment]
I agree - "Automatic Configuration" makes more sense for naming. This would mean the default value would be `true` and we'd need to backfill. If this checks out with you it works for me.

**Comment by jlee9595** [comment]
Yeah that sounds perfect and I believe that's how we've handled these kinds of situations in the past

**Comment by jlee9595** [comment]
Looking at the FB docs looks like there's also the option to send multiple content_types. Is that something that makes sense for us to support now or does it make more sense to punt on it till a customer needs it?

File: integrations/facebook-pixel/lib/index.js [hunk 1]
```diff
@@ -52,6 +54,9 @@ FacebookPixel.prototype.initialize = function() {
   window.fbq.version = '2.0';
   window.fbq.queue = [];
   this.load(this.ready);
+  if (!this.options.automaticConfiguration) {
```
Comment (gpsamson) [inline integrations/facebook-pixel/lib/index.js]: This conditional is relevant to [PLATFORM-2310](https://segment.atlassian.net/browse/PLATFORM-2310).

All changes below this belong to [PLATFORM-2538](https://segment.atlassian.net/browse/PLATFORM-2538)

File: integrations/facebook-pixel/test/index.test.js [hunk 2]
```diff
@@ -86,6 +90,20 @@ describe('Facebook Pixel', function() {
         analytics.assert(window.fbq instanceof Function);
       });
 
+      before(function() {
```
Comment (ccnixon) [inline integrations/facebook-pixel/test/index.test.js]: I would omit these `before` and `after` blocks and just explicitly toggle this option before and after the test code runs. So like this:

```
      it('should call set autoConfig if option disableAutoConfig is enabled', function () {
options. automaticConfiguration = false
        analytics.stub(window, 'fbq');
        analytics.initialize();
        analytics.called(window.fbq, 'set', 'autoConfig', false, options.pixelId);
options.automaticConfiguration = true;
      });
```

Comment (ccnixon) [reply integrations/facebook-pixel/test/index.test.js]: Unless you think there would be a need for future tests, in which case, I would use `beforeEach` and `afterEach`.

Comment (gpsamson) [reply integrations/facebook-pixel/test/index.test.js]: Ahh, doing it in the test code doesn't work since a.js is initialized before-hand.

**Pull request merged**